controller/space_iterations_blackbox_test.go

```diff
@@ -0,0 +1,558 @@
+	"github.com/fabric8-services/fabric8-wit/iteration"
+	"github.com/fabric8-services/fabric8-wit/ptr"
+	"github.com/fabric8-services/fabric8-wit/resource"
+	"github.com/fabric8-services/fabric8-wit/space"
+	"github.com/fabric8-services/fabric8-wit/spacetemplate"
+	testsupport "github.com/fabric8-services/fabric8-wit/test"
+	tf "github.com/fabric8-services/fabric8-wit/test/testfixture"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type SpaceIterationControllerTestSuite struct {
+	gormtestsupport.DBTestSuite
+	db           *gormapplication.GormDB
+	testIdentity account.Identity
+	testDir      string
+}
+
+func TestSpaceIterationController(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &SpaceIterationControllerTestSuite{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+func (s *SpaceIterationControllerTestSuite) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.db = gormapplication.NewGormDB(s.DB)
+	testIdentity, err := testsupport.CreateTestIdentity(s.DB, "SpaceIterationControllerTestSuite user", "test provider")
+	require.NoError(s.T(), err)
+	s.testIdentity = *testIdentity
+	req := &http.Request{Host: "localhost"}
+	params := url.Values{}
+	s.Ctx = goa.NewContext(context.Background(), nil, req, params)
+	s.testDir = filepath.Join("test-files", "space_iterations")
+}
+
+func (s *SpaceIterationControllerTestSuite) SecuredController(identity ...account.Identity) (*goa.Service, *SpaceIterationsController) {
+	i := testsupport.TestIdentity
+	if identity != nil && len(identity) > 0 {
+		i = identity[0]
+	}
+	svc := testsupport.ServiceAsUser("Iteration-Service", i)
+	return svc, NewSpaceIterationsController(svc, s.db, s.Configuration)
+}
+
+func (s *SpaceIterationControllerTestSuite) SecuredControllerWithIdentity(idn *account.Identity) (*goa.Service, *SpaceIterationsController) {
+	svc := testsupport.ServiceAsUser("Iteration-Service", *idn)
+	return svc, NewSpaceIterationsController(svc, s.db, s.Configuration)
+}
+
+func (s *SpaceIterationControllerTestSuite) UnSecuredController() (*goa.Service, *SpaceIterationsController) {
+	svc := goa.New("Iteration-Service")
+	return svc, NewSpaceIterationsController(svc, s.db, s.Configuration)
+}
+
+func (s *SpaceIterationControllerTestSuite) TestCreate() {
+	s.T().Run("success", func(t *testing.T) {
+		t.Run("ok", func(t *testing.T) {
+			// given
+			ci := newCreateSpaceIterationPayload("Sprint #42", nil)
+			fxt := tf.NewTestFixture(t, s.DB, tf.CreateWorkItemEnvironment())
+			svc := testsupport.ServiceAsUser("Iteration-Service", *fxt.Identities[0])
+			ctrl := NewSpaceIterationsController(svc, s.db, s.Configuration)
+			// when
+			resp, iter := test.CreateSpaceIterationsCreated(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
```

Comment:
Can you please also add a golden file comparison for the creation payload right before this line? So something like this:

```go
compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "ok.payload.req.golden.json"), ci)